filter/ArticleMedraXmlFilter.inc.php

```diff
@@ -144,6 +144,80 @@ function createArticleNode($doc, $pubObject) {
 			$url = PKPString::regexp_replace('#://[^\s]+/index.php#', '://example.com/index.php', $url);
 		}
 		$articleNode->appendChild($node = $doc->createElementNS($deployment->getNamespace(), 'DOIWebsiteLink', $url));
+
+		// Adding Collection on the content 
+		//Collection property="crawler-based"
+		$galleys = $pubObject->getData('galleys');
+		// All full-texts, PDF full-texts and remote galleys for text-mining and as-crawled URL
+		$submissionGalleys = $pdfGalleys = $remoteGalleys = array();
+		// preferred PDF full-text for the as-crawled URL
+		$pdfGalleyInArticleLocale = null;
+		$genreDao = DAORegistry::getDAO('GenreDAO'); /* @var $genreDao GenreDAO */
+		foreach ($galleys as $gal) {
+			if (!$gal->getRemoteURL()) {
+				$galleyFile = $gal->getFile();
+				if ($galleyFile) {
+					$genre = $genreDao->getById($galleyFile->getGenreId());
+					if (!$genre->getSupplementary()) {
+						$submissionGalleys[] = $gal;
+						if ($gal->isPdfGalley()) {
+							$pdfGalleys[] = $gal;
+							if (!$pdfGalleyInArticleLocale && $gal->getLocale() == $locale) {
+								$pdfGalleyInArticleLocale = $gal;
+							}
+						}
+					}
+				}
+			} else {
+				$remoteGalleys[] = $gal;
+			}
+		}	
+			
+		// as-crawled URLs
+		$asCrawledGalleys = array();
+		if ($pdfGalleyInArticleLocale) {
+			$asCrawledGalleys = array($pdfGalleyInArticleLocale);
+		} elseif (!empty($pdfGalleys)) {
+			$asCrawledGalleys = array($pdfGalleys[0]);
+		} else {
+			$asCrawledGalleys = $submissionGalleys;
+		}
+		foreach ($asCrawledGalleys as $crawledGalley) {
```

Comment:
could you put it into a separate function, similar to this here: https://github.com/pkp/ojs/blob/stable-3_3_0/plugins/importexport/crossref/filter/ArticleCrossrefXmlFilter.inc.php#L280 ?